src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -945,31 +1001,15 @@ private Task WriteToStreamAsync(ReadOnlyMemory<byte> source, CancellationToken c
                 _readLength = remaining;
             }
 
-            // When the connection was put back into the pool, a pre-emptive read was performed
-            // into the read buffer.  That read should not complete prior to us using the
-            // connection again, as that would mean the connection was either closed or had
-            // erroneous data sent on it by the server in response to no request from us.
-            // We need to consume that read prior to issuing another read request.
-            Task<int> t = _readAheadTask;
-            int bytesRead;
-            if (t != null)
-            {
-                Debug.Assert(_readOffset == 0);
-                Debug.Assert(_readLength == 0);
-                _readAheadTask = null;
-                bytesRead = await t.ConfigureAwait(false);
-            }
-            else
-            {
-                ValueTask<int> vt = _stream.ReadAsync(new Memory<byte>(_readBuffer, _readLength, _readBuffer.Length - _readLength), cancellationToken);
-                bytesRead = vt.IsCompletedSuccessfully ? vt.Result : await vt.AsTask().ConfigureAwait(false);
-            }
+            ValueTask<int> vt = _stream.ReadAsync(new Memory<byte>(_readBuffer, _readLength, _readBuffer.Length - _readLength), cancellationToken);
+            int bytesRead = vt.IsCompletedSuccessfully ? vt.Result : await vt.AsTask().ConfigureAwait(false);
```

Comment:
As long as you're changing this, I was "too clever" here, and with some other changes being contemplated, this could actually make things worse.  Better to just do:
```C#
int bytesRead = await vt.ConfigureAwait(false);
```
or combining the await into the previous line.